confidential/confidential.go

```diff
@@ -22,19 +22,19 @@ type UnblindOutputResult struct {
 }
 
 //NonceHash method generates hashed secret based on ecdh
-func NonceHash(ctx *secp256k1.Context, pubKey, privKey []byte) (*[32]byte, error) {
+func NonceHash(ctx *secp256k1.Context, pubKey, privKey []byte) ([32]byte, error) {
```

Comment:
It would be more readable if  we name the return variables here, so we can avoid to return a `[32]byte{}` when an error occurs:
```suggestion
func NonceHash(ctx *secp256k1.Context, pubKey, privKey []byte) (result [32]byte, err error) {
```

Reply:
Done